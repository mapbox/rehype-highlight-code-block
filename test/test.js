'use strict';

const rehype = require('rehype');
const dedent = require('dedent');
const rehypeHighlightCodeBlock = require('..');

describe('rehypeHighlightCodeBlock', () => {
  let mockHighLightSpy;
  let processHtml;

  beforeEach(() => {
    mockHighLightSpy = jest.fn(() => '<span>mockSyntaxHighlighting</span>');
    processHtml = html => {
      return rehype()
        .data('settings', { fragment: true })
        .use(rehypeHighlightCodeBlock, {
          highlight: mockHighLightSpy
        })
        .processSync(html)
        .toString();
    };
  });

  test('mocks syntax highlighting with language-* class', () => {
    const result = processHtml(dedent`
      <div>
        <p>foo</p>
        <pre><code class="language-css">p { color: red }</code></pre>
        <p>bar</p>
      </div>
    `);
    expect(mockHighLightSpy.mock.calls[0].length).toBe(2);
    expect(mockHighLightSpy.mock.calls[0][1]).toBe('css');
    expect(result).toMatchSnapshot();
  });

  test('mocks syntax highlighting with lang-* class', () => {
    const result = processHtml(dedent`
      <div>
        <p>foo</p>
        <pre><code class="lang-css">p { color: red }</code></pre>
        <p>bar</p>
      </div>
    `);
    expect(mockHighLightSpy.mock.calls[0].length).toBe(2);
    expect(mockHighLightSpy.mock.calls[0][1]).toBe('css');
    expect(result).toMatchSnapshot();
  });

  test('mocks syntax highlighting with no language class', () => {
    const result = processHtml(dedent`
      <div>
        <p>foo</p>
        <pre><code>p { color: red }</code></pre>
        <p>bar</p>
      </div>
    `);
    expect(mockHighLightSpy.mock.calls[0].length).toBe(2);
    expect(mockHighLightSpy.mock.calls[0][1]).toBe(null);
    expect(result).toMatchSnapshot();
  });

  test('does nothing to code block, missing <pre></pre>', () => {
    const result = processHtml(dedent`
      <code>p { color: red }</code>
    `);
    expect(mockHighLightSpy.mock.calls.length).toBe(0);
    expect(result).toMatchSnapshot();
  });

  test('does nothing to code block, missing <code></code>', () => {
    const result = processHtml(dedent`
      <pre>p { color: red }</pre>
    `);
    expect(mockHighLightSpy.mock.calls.length).toBe(0);
    expect(result).toMatchSnapshot();
  });

  test('does nothing to code block, missing pre > code', () => {
    const result = processHtml(dedent`
      <div>p { color: red }</div>
    `);
    expect(mockHighLightSpy.mock.calls.length).toBe(0);
    expect(result).toMatchSnapshot();
  });
});
