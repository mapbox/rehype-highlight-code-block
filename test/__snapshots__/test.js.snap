// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rehypeHighlightCodeBlock does nothing to code block, missing <code></code> 1`] = `"<pre>p { color: red }</pre>"`;

exports[`rehypeHighlightCodeBlock does nothing to code block, missing <pre></pre> 1`] = `"<code>p { color: red }</code>"`;

exports[`rehypeHighlightCodeBlock does nothing to code block, missing pre > code 1`] = `"<div>p { color: red }</div>"`;

exports[`rehypeHighlightCodeBlock mocks syntax highlighting with lang-* class 1`] = `
"<div>
  <p>foo</p>
  <pre><code class=\\"lang-css\\"><span>mockSyntaxHighlighting</span></code></pre>
  <p>bar</p>
</div>"
`;

exports[`rehypeHighlightCodeBlock mocks syntax highlighting with language-* class 1`] = `
"<div>
  <p>foo</p>
  <pre><code class=\\"language-css\\"><span>mockSyntaxHighlighting</span></code></pre>
  <p>bar</p>
</div>"
`;

exports[`rehypeHighlightCodeBlock mocks syntax highlighting with no language class 1`] = `
"<div>
  <p>foo</p>
  <pre><code><span>mockSyntaxHighlighting</span></code></pre>
  <p>bar</p>
</div>"
`;
